src/widget/wfindonmenu.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <QAction>
+#include <QMenu>
+#include <QUrlQuery>
+
+class Track;
+
+class WFindOnMenu : public QMenu {
+    Q_OBJECT
+  public:
+    explicit WFindOnMenu(
+            QWidget* parent = nullptr);
+    ~WFindOnMenu() override = default;
+
+    enum class Service {
+        SoundCloud,
+        LastFm,
+        Discogs
+    };
+
+    enum class TrackProperty {
+        Artist,
+        ArtistTitle,
+        ArtistAlbum,
+        Album,
+        Title
+    };
+
+  public slots:
+    void openInBrowser(WFindOnMenu::Service service,
+            WFindOnMenu::TrackProperty trackProperty,
+            const QString& query);
+
+    void createAllSubmenusForWebLookups(const Track& track);
```

Comment:
Why does this need to be a public slot? It is invoked as an ordinary member function.

Reply:
"find on", "web lookups", now we have already 2 different terms for the same thing. The name of the member functions should also not redundantly repeat the purpose of the class.

Imho a name like `populateFromTrackProperties()` would be more suitable here. That this task involves sub menus is an implementation detail that should not be revealed and doesn't matter for the caller.

Reply:
> Why does this need to be a public slot? It is invoked as an ordinary member function.

I was calling these functions to connect signals in WTrackMenu.

> The name of the member functions should also not redundantly repeat the purpose of the class.

I will be also careful about that too.  

> Imho a name like populateFromTrackProperties() would be more suitable here. That this task involves sub menus is an implementation detail that should not be revealed and doesn't matter for the caller.

This function is generating submenus named with the "services-websites" and it is called in WTrackMenu. IMHO, I would like to use "addMenusForWebSearch". Since it is called in WTrackMenu, that would be easier to understand what this function does.

The naming also can be createAllMenusForWebSites or createMenuForServices. I think we should decide it later on what this feature will be called.

BTW this function "createAllSubmenusForWebLookups" is calling another functions right now, they are named "populateWebLookUpQueries", this can be named as "populateFromTrackProperties" which is more like they are intended to do. 

What do you think?

Reply:
I realized that I am repeating the purpose of the class by using "addMenusForWebSearch". "WebSearch" part is revealing the purpose.

Since I named the websites as Service. This function creates QMenu named with service names. Should I make it something like "addSubmenusForServices"?